src/clonemap.rs

```diff
@@ -0,0 +1,859 @@
+    ///
+    /// This acts the same as [insert][CloneConMap::insert], but takes
+    /// the already created element. It can be used when:
+    ///
+    /// * `V: ?Sized`.
+    /// * You want to insert the same element into multiple maps.
+    pub fn insert_element(&self, element: Element<K, V>) -> Option<Element<K, V>> {
+        let pin = crossbeam_epoch::pin();
+        self.raw
+            .insert(CloneMapPayload(element), &pin)
+            .map(|p| (p.0).clone())
+    }
+
+    /// Looks up or inserts a new element.
+    ///
+    /// This is the same as
+    /// [get_or_insert_with][CloneConMap::get_or_insert_with], but the
+    /// closure returns a pre-created element. This can be used when:
+    ///
+    /// * `V: ?Sized`.
+    /// * You want to insert the same element into multiple maps.
+    pub fn get_or_insert_with_element<F>(&self, key: K, create: F) -> ExistingOrNew<Element<K, V>>
+    where
+        F: FnOnce(K) -> Element<K, V>,
+    {
+        let pin = crossbeam_epoch::pin();
+        self.raw
+            .get_or_insert_with(key, |key| CloneMapPayload(create(key)), &pin)
+            .map(|payload| (payload.0).clone())
+    }
+
+    /// Looks up an element.
+    pub fn get<Q>(&self, key: &Q) -> Option<Element<K, V>>
+    where
+        Q: ?Sized + Eq + Hash,
+        K: Borrow<Q>,
+    {
+        let pin = crossbeam_epoch::pin();
+        self.raw.get(key, &pin).map(|r| (r.0).clone())
+    }
+
+    /// Removes an element identified by the given key, returning it.
+    pub fn remove<Q>(&self, key: &Q) -> Option<Element<K, V>>
+    where
+        Q: ?Sized + Eq + Hash,
+        K: Borrow<Q>,
+    {
+        let pin = crossbeam_epoch::pin();
+        self.raw.remove(key, &pin).map(|r| (r.0).clone())
+    }
+}
+
+impl<K, V, S> CloneConMap<K, V, S>
+where
+    K: Clone + Hash + Eq,
+    V: Clone,
+{
+    /// Checks if the map is currently empty.
+    ///
+    /// Note that due to the nature of concurrent map, this is
+    /// inherently racy ‒ another thread may add or remove elements
+    /// between you call this method and act based on the result.
+    pub fn is_empty(&self) -> bool {
+        self.raw.is_empty()
+    }
+
+    /// Returns an iterator through the elements of the map.
+    pub fn iter(&self) -> Iter<K, V, S> {
+        Iter {
+            inner: raw::iterator::Iter::new(&self.raw),
+        }
+    }
+}
+
+impl<K, V> Default for CloneConMap<K, V>
+where
+    K: Clone + Hash + Eq,
+    V: Clone,
+{
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl<K, V, S> Debug for CloneConMap<K, V, S>
+where
+    K: Debug + Clone + Hash + Eq,
+    V: Debug + Clone,
+{
+    fn fmt(&self, fmt: &mut Formatter) -> FmtResult {
+        let mut d = fmt.debug_map();
+        for n in self {
+            // Hack: As of 1.37.0 the parameters to entry need to be
+            // &Sized. By using a double-ref, we satisfy the compiler's
+            // pickiness in that regard.
+            let val: &&V = &n.value();
```

Comment:
Again, this won't work for unsized `V`s anyway, so there's little use for this hack.